contracts/starknet/space/space.cairo

```diff
@@ -52,92 +67,221 @@ end
+        voting_params_len : felt, voting_params : felt*) -> (voting_power : Uint256):
+    alloc_locals
+    # Get voting strategy contract
+    let (voting_strategy_contract) = voting_strategies.read(index)
+
+    if voting_strategy_contract == 0:
+        # Reached the end, stop iteration
+        return (Uint256(0, 0))
+    end
+
+    let (user_voting_power) = IVotingStrategy.get_voting_power(
+        contract_address=voting_strategy_contract,
+        timestamp=current_timestamp,
+        address=voter_address,
+        params_len=voting_params_len,
+        params=voting_params)
+    let (additional_voting_power) = get_cumulated_voting_power(
+        index + 1, current_timestamp, voter_address, voting_params_len, voting_params)
+
+    let (voting_power, overflow) = uint256_add(user_voting_power, additional_voting_power)
+    with_attr error_message("Overflow while computing voting power"):
+        if overflow != 0:
+            # Overflow happened, revert transaction
+            assert 1 = 0
+        end
+    end
+
+    return (voting_power)
+end
+
 @constructor
 func constructor{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr : felt}(
         _voting_delay : felt, _voting_period : felt, _proposal_threshold : Uint256,
-        _voting_strategy : felt, _authenticator : felt):
+        _executor : felt, _controller : felt, _voting_strategies_len : felt,
+        _voting_strategies : felt*, _authenticators_len : felt, _authenticators : felt*):
     # Sanity checks
-    assert_nn(_voting_delay)
-    assert_nn(_voting_period)
-    assert_not_zero(_voting_strategy)
-    assert_not_zero(_authenticator)
+    with_attr error_message("Invalid constructor parameters"):
+        assert_nn(_voting_delay)
+        assert_nn(_voting_period)
+        assert_not_zero(_executor)
+        assert_not_zero(_controller)
+        assert_not_zero(_voting_strategies_len)
+        assert_not_zero(_authenticators_len)
+    end
     # TODO: maybe use uint256_signed_nn to check proposal_threshold?
+    # TODO: maybe check that _executor is not 0?
 
     # Initialize the storage variables
     voting_delay.write(_voting_delay)
     voting_period.write(_voting_period)
     proposal_threshold.write(_proposal_threshold)
-    voting_strategy.write(_voting_strategy)
-    authenticator.write(_authenticator)
+    executor.write(_executor)
+    controller.write(_controller)
+
+    register_voting_strategies(0, _voting_strategies_len, _voting_strategies)
+    register_authenticators(_authenticators_len, _authenticators)
+
     next_proposal_nonce.write(1)
 
     return ()
 end
 
 @external
 func vote{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr : felt}(
-        voter_address : EthAddress, proposal_id : felt, choice : felt, params_len : felt,
-        params : felt*) -> ():
+        voting_strategy_contract : felt, voter_address : EthAddress, proposal_id : felt,
```

Comment:
`voting_strategy_contract` shouldnt be passed here

Reply:
hehe indeed, dead code :) 908226416fa83d9f210c2182f00c6e5bcbe665ac